### PR TITLE
Feature/pip

### PR DESCRIPTION
**Pull request by a4z**
This creates a python pip that can be uploaded

Ready to test: 
https://pypi.org/project/djinni/1.2.0/

As you can see at the type in the README, something I am famous for, this part should become an own project and not be part of the support lib, since I think they change mostly independent from each other.

Anyhow, you can test the pip by install it,  removing `"${CMAKE_SOURCE_DIR}/djinni/py/src"` from the python path, and build and run the test.

So the POC works, we will have to see how we maintain that in the future

### CODE REVIEW

**Comment by jothepro** [comment]
To avoid confusion maybe it would be better to name the pip-dependency 'djinni-support-lib' instead of 'djinni', to make sure that it's not the generator? 🤔

**Comment by a4z** [comment]
there is not djinni python generator
the usage from python is
`import djinni` or `from djinni import exception` 
beside that there is no confusion,
I do not think it will be beneficial when we change the name to something long no one wants to type ;-)

**Review by freitass — APPROVED** [review]
> As you can see at the type in the README, something I am famous for, this part should become an own project and not be part of the support lib, since I think they change mostly independent from each other.

I'm not sure I follow that comment. Do you mean that files such as `exception.py`, `pycffi_marshal.py` and `support.py` should not be coupled with the support-lib project?

Other than that, changes look good! 👍

**Comment by a4z** [comment]
Sorry for creating confusion Leandro.

In the first version of the README, that went on https://pypi.org/project/djinni/1.2.0/,  I have written *clue* glue .... 
> This is the required python **clue** to make djinni generated python bindings work.

but that should be

> This is the required python **glue** to make djinni generated python bindings work.

Fixing that requires a new version on pip ..... 

But we might not only want to  fix typos in readme, but in general, not having a new version when a new support lib is release, or fixing something in the pip with out releasing a new support lib version.

This is why I think the pip part should have it's extra repo.

**Comment by a4z** [comment]
fyi, I am considering committing that soon, since it works and the case of extra pip releases is currently not given.
so I expect the current published pip work for some time (support lib versions)
for the rare case  we seriously break the interface , the manual step of releasing the pip is not an issue.

if there are any considerations, pease let me know .

**Review by jothepro — COMMENTED** [review]
- While I understand that it shouldn't be too hard to update the pip manually, I'd prefer if there was automation for it so any maintainer could in theory do a release without having to know the details and the password. But that's not a show stopper for me, we could still add this afterwards. 👍 
- Did you publish this via a private account? If we plan to host this repo in this organization, I'd prefer if we published the package with an organization account and if the package would have our organization-email in the metadata, both to reduce the risk of having a single point of failure.
- To make sure I'm not missing sth: As far as I can tell the changes in CMakeListst.txt are just cosmetic, is that correct? The pip would work with or without them?
- Afaik you are proposing to make this a separate repository with separate versioning? How is it going to pull the sources? Are you suggesting to move all python sources over to the other repository, with a separate versioning lifecycle?

File: djinni/py/setup.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import setuptools
+
+with open("README.md", "r", encoding="utf-8") as fh:
+    long_description = fh.read()
+
+setuptools.setup(
+    name="djinni",
+    version="1.2.1",
```
[the remark this answers is not in the document]
Comment (jothepro) [reply djinni/py/setup.py:8]: Nevermind, I realized that your are suggesting to let this have it's own versioning.

**Comment by a4z** [comment]
> * While I understand that it shouldn't be too hard to update the pip manually, I'd prefer if there was automation for it so any maintainer could in theory do a release without having to know the details and the password. But that's not a show stopper for me, we could still add this afterwards. +1

Automation can happen once we know how the infrastructure will look like.
As mentioned, I think the pip source might be well suited in an extra repo, since it is kind of independent from the support lib as long as the support lib does not do breaking changes. But this is just 1 possibility. We could also use different tags and going via names of them, .... and there are other ways also .


> * Did you publish this via a private account? If we plan to host this repo in this organization, I'd prefer if we published the package with an organization account and if the package would have our organization-email in the metadata, both to reduce the risk of having a single point of failure.

For now yes since I needed to start with something, but I plan of course to change that so this can be done via github.
But for this we need to know how the infrastructure will look like so I can add keys.

> * To make sure I'm not missing sth: As far as I can tell the changes in CMakeListst.txt are just cosmetic, is that correct? The pip would work with or without them?

The pip has no relation to the CMake file. 
The Python test build via cmake requires the pip installed. It's basically testing the pip, and the support lib python bindings.
The pip can be pulled from pip or, as long as it is like it is onw, installed locally from the pip sources in this repo. 
But it simulates already a uses who has the pip installed.

> * Afaik you are proposing to make this a separate repository with separate versioning? How is it going to pull the sources? Are you suggesting to move all python sources over to the other repository, with a separate versioning lifecycle?

This would be one possibility.  But there are multiple.
It's definitely so that there is no relation between pip and support lib, except when there is (we make breaking changes in support lib that requires a new pip) 
But we can have releases of either, for example adding improvements,  without the need to update the other

**Comment by a4z** [comment]
Turned out in it's current form this is not useful. 
And it's not clear if shipping this source via pip is wanted, or if the required code should be generated by the generator.
Very likely that should come from the generator.

**Pull request closed without merging**